executing/executing.py

```diff
@@ -509,9 +530,556 @@ def visit_ClassDef(self, node):
+                # This is save, because there can only be one attribute ending at a specific point in the source code.
+                node = self.find_node(
+                lasti,
+                match_positions=("end_col_offset", "end_lineno"),
+                typ=ast.Attribute,
+            )
+            elif self.opname(lasti) == "CALL":
+                # A CALL instruction can be a method call, in which case the lineno and col_offset gets changed by the compiler.
+                # Therefore we ignoring here this attributes and searchnig for a Call-node only by end_col_offset and end_lineno. 
+                # This is save, because there can only be one method ending at a specific point in the source code.
+                # One closing ) only belongs to one method.
+                node = self.find_node(
+                    lasti,
+                    match_positions=("end_col_offset", "end_lineno"),
+                    typ=ast.Call,
+                )
+            else:
+                raise
+
+        # report KnownIssues
+
+        instruction = self.instruction(lasti)
+
+        if instruction.opname in ("COMPARE_OP", "IS_OP", "CONTAINS_OP") and isinstance(
+            node, self.types_cmp_issue
+        ):
+            if isinstance(node, self.types_cmp_issue_fix):
+                # this is a workaround for https://github.com/python/cpython/issues/95921
+                # we can fix cases with only on comparison inside the test condition
+                #
+                # we can not fix cases like:
+                # if a<b<c and d<e<f: pass
+                # if (a<b<c)!=d!=e: pass
+                # because we don't know which comparison caused the problem
+
+
+                comparisons = [
+                    n
+                    for n in ast.walk(node.test)
+                    if isinstance(n, ast.Compare) and len(n.ops) > 1
+                ]
+
+                assert_(comparisons, "expected at least one comparison")
+
+                if len(comparisons) == 1:
+                    node = comparisons[0]
+                else:
+                    raise KnownIssue(
+                        "multiple chain comparison inside %s can not be fixed" % (node)
+                    )
+
+            else:
+                # Comprehension and generators get not fixed for now.
+                raise KnownIssue("chain comparison inside %s can not be fixed" % (node))
+
+        if isinstance(node, ast.Assert):
+            # pytest assigns the position of the assertion to all expressions of the rewritten assertion.
+            # All the rewritten expressions get mapped to ast.Assert, which is the wrong ast-node.
+            # We don't report this wrong result.
+            raise KnownIssue("assert")
+
+        if any(isinstance(n, ast.pattern) for n in node_and_parents(node)):
+            # TODO: investigate
+            raise KnownIssue("pattern matching ranges seems to be wrong")
```

Comment:
I have not investigated this issue and not even the cases I found.

I only remember some wired positions, but it might even be that everything works as expected and it was only the pattern matching that confused me. Is pattern matching support important? Does the current version of *executing* support it?

Reply:
No it's not important. If you had actually clearly observed a problem then I think it would be worth reporting it (and hopefully someone else would fix it), but since you haven't actually nailed down a specific problem then just leave it.